### PR TITLE
Clarify empty/0 values

### PR DESCRIPTION
**Pull request by tdraier**
fixes: https://github.com/dust-tt/tasks/issues/8819

## Description

Clarifies the meaning of empty/zero values across programmatic and pool credit settings, and fixes related logic bugs.

### UI / UX

- **Programmatic monthly limit**: `null` or `0` → empty field with "No access" placeholder; "credits" unit label hidden until focused
- **Workspace credit pool threshold**: `0` → empty field with "Disabled" placeholder; "credits" unit label hidden until focused
- **Default workspace credit pool limit**: `null` or `0` → empty field with "No access" placeholder; "credits" unit label hidden until focused
- **Notification copy**: saving a limit of 0 now says "Programmatic access disabled" instead of "Programmatic usage limit has been removed"

### Logic / bug fixes

- **Empty field saves 0, not null**: clearing the programmatic limit field now persists `0` (hard block) rather than `null` (no limit), matching the intent of the placeholder text
- **Alerts skipped for cap = 0**: a cap of 0 is always fully depleted, so no threshold transition can ever fire; Metronome alerts are now cleared (not upserted) in that case
- **Immediate state reconciliation**: after saving a cap change, `programmaticCreditState` is reconciled right away so `/usage-status` reflects the new state without waiting for a Metronome webhook
- **Reconcile reads DB first for cap = 0**: the reconciliation logic now reads the stored cap before querying Metronome alert states; Metronome returns "active" when no alerts exist, which would incorrectly override the "depleted" state for a cap of 0
- **Banner null-safety**: `AppStatusBanner` now coalesces `monthlyCapCredits ?? 0` before comparing to `0`, so a `null` cap is treated as a hard block
- **SWR cache invalidation**: `useUpdateProgrammaticUsageLimit` now also mutates `/usage-status` after a successful save so the banner updates immediately

## Tests

Unit tests updated for the cap = 0 alert behavior. Tested manually.

## Risk

Low — UI is display-only; logic changes only affect the 0-cap edge case which had no real contracts yet.

## Deploy Plan

Standard deploy, no migration needed.

### CODE REVIEW

**Comment by tdraier** [comment]
This change is part of the following stack:

- #27591 ◀
    - #27618

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

File: front/components/workspace/usage/UsageProgrammaticLimitCard.tsx [hunk 1]
```diff
@@ -21,12 +22,14 @@ export function UsageProgrammaticLimitCard({
...
+  const [isEditing, setIsEditing] = useState(false);
+
   const handleSaveLimit = async (newValue: string) => {
     const trimmed = newValue.trim();
 
-    // An empty value means no limit.
+    // An empty value means no limit (same as null or 0).
     if (trimmed === "") {
-      if (currentLimit === null) {
+      if (currentLimit === null || currentLimit === 0) {
```
Comment (thervier) [inline front/components/workspace/usage/UsageProgrammaticLimitCard.tsx:33]: by reading this, I have the feeling that one could want to set a limit to 0 that does not correspond to "no limit" but actually the opposite, to disable programmatic usage. Might be also the case for the others ?

Comment (tdraier) [reply front/components/workspace/usage/UsageProgrammaticLimitCard.tsx:33]: Right, no programmatic usage at all can be useful

Comment (tdraier) [reply front/components/workspace/usage/UsageProgrammaticLimitCard.tsx:33]: Ok so actually 0 should mean programmatic disabled - no limit is not an option.

**Review by thervier — APPROVED** [review]
LGTM (maybe we just need an update of the PR description to fit, for future checks by humans or agents :) )
